tls/_constructs.py

```diff
@@ -67,6 +68,25 @@
     Array(lambda ctx: ctx.length, UBInt8("compression_methods"))
 )
 
+HostName = Struct(
```

Comment:
From @ashfall's [comment on this PR](https://github.com/pyca/tls/pull/105#issuecomment-261179286):

> I'm not very happy about doing server_name_list[0].name.data and would rather 'localhost' be server_name_list[0].name, i.e. not have the HostName container as a thing.

I think you can remove the `Struct` and name the `PrefixedBytes` `hostname`:

```python
HostName = PrefixedBytes("hostname", UBInt16("length"))
```

But I haven't tested it.